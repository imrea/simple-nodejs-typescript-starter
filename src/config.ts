export const PROD = process.env.NODE_ENV === 'production';
export const DEV = process.env.NODE_ENV === 'development';
export const TEST = process.env.NODE_ENV === 'test';

export const PORT = Number(process.env.PORT) || 3000;
