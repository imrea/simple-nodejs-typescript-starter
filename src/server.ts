import Koa from 'koa';

const app = new Koa();

app.use((ctx) => {
  ctx.body = { message: 'Hello World!' };
});

export { app };
