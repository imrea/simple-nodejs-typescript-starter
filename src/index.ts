import { PORT } from './config';
import { app } from './server';

async function main() {
  app.listen(PORT, () => {
    console.log(`Server is listening on 0.0.0.0:${PORT}`);
  });
}

main();
