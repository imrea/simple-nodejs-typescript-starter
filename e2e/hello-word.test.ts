import request from 'supertest';
import { app } from '../src/server';

describe('hello world', () => {
  const server = app.callback();

  it('returns Hello World!', async () => {
    const response = await request(server).get('/');
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});
