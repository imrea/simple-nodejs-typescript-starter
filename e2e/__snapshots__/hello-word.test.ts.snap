// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello world returns Hello World! 1`] = `
Object {
  "message": "Hello World!",
}
`;
